src/control/cmd/daos/util.go

```diff
@@ -138,24 +138,23 @@ func freeString(str *C.char) {
 	C.free(unsafe.Pointer(str))
 }
 
-func createWriteStream(ctx context.Context, prefix string, printLn func(line string)) (*C.FILE, func(), error) {
-	// Create a FILE object for the handler to use for
-	// printing output or errors, and call the callback
-	// for each line.
+func createWriteStream(ctx context.Context, printLn func(line string)) (*C.FILE, func(), error) {
+	// Create a FILE object for the handler to use for printing output or errors, and call the
+	// callback for each line.
 	r, w, err := os.Pipe()
 	if err != nil {
 		return nil, nil, err
 	}
+	var done = new(sync.Mutex)
+	done.Lock()
 
 	stream, err := fd2FILE(w.Fd(), "w")
 	if err != nil {
 		return nil, nil, err
 	}
 
-	go func(ctx context.Context, prefix string) {
-		if prefix != "" {
-			prefix = ": "
-		}
+	go func(ctx context.Context) {
+		defer done.Unlock()
```

Comment:
normal pattern is to lock before deferring unlock, with this don't you risk unlocking multiple times?

Reply:
The lock is locked as it's created, and remains locked until this function (thread?) returns so I don't see how it would be unlocked multiple times?